SpeechResponder/ScriptResolver.cs

```diff
@@ -1083,28 +1083,42 @@ CommodityMarketQuote CommodityDetails(string commodityLocalizedName, Station sta
-                Cottle.Value value_0 = values[0];
-                Cottle.Value value_1 = values[1];
-                if (values.Count == 2 && value_0.Type == Cottle.ValueContent.String && value_1.Type == Cottle.ValueContent.String)
+                bool numVal = values[0].Type == Cottle.ValueContent.Number;
+                bool stringVal = values[0].Type == Cottle.ValueContent.String;
+
+                StarSystem curr = new StarSystem();
+                StarSystem dest = new StarSystem();
+                if (values.Count == 1 && stringVal)
                 {
-                    StarSystem curr = StarSystemSqLiteRepository.Instance.GetOrFetchStarSystem(values[0].AsString, true);
-                    StarSystem dest = StarSystemSqLiteRepository.Instance.GetOrFetchStarSystem(values[1].AsString, true);
-                    if (curr?.x != null && dest?.x != null)
-                    {
-                        result = (decimal)Math.Round(Math.Sqrt(Math.Pow((double)(curr.x - dest.x), 2)
-                                    + Math.Pow((double)(curr.y - dest.y), 2)
-                                    + Math.Pow((double)(curr.z - dest.z), 2)), 2);
-                    }
+                    curr = EDDI.Instance?.CurrentStarSystem;
+                    dest = StarSystemSqLiteRepository.Instance.GetOrFetchStarSystem(values[0].AsString, true);
+                }
+                else if (values.Count == 2 && stringVal)
+                {
+                    curr = StarSystemSqLiteRepository.Instance.GetOrFetchStarSystem(values[0].AsString, true);
+                    dest = StarSystemSqLiteRepository.Instance.GetOrFetchStarSystem(values[1].AsString, true);
                 }
-                else if (values.Count == 6 && value_0.Type == Cottle.ValueContent.Number && value_1.Type == Cottle.ValueContent.Number)
+                else if (values.Count == 6 && numVal)
                 {
-                    result = (decimal)Math.Round(Math.Sqrt(Math.Pow((double)(values[0].AsNumber - values[3].AsNumber), 2)
-                                + Math.Pow((double)(values[1].AsNumber - values[4].AsNumber), 2)
-                                + Math.Pow((double)(values[2].AsNumber - values[5].AsNumber), 2)), 2);
+                    curr.x = values[0].AsNumber;
+                    curr.y = values[1].AsNumber;
+                    curr.z = values[2].AsNumber;
+                    dest.x = values[3].AsNumber;
+                    dest.y = values[4].AsNumber;
+                    dest.z = values[5].AsNumber;
                 }
+
+                if (curr?.x != null && dest?.x != null)
+                {
+                    result = (decimal)Math.Round(Math.Sqrt(square((double)(curr.x - dest.x))
+                                + square((double)(curr.y - dest.y))
+                                + square((double)(curr.z - dest.z))), 2);
+                }
+
                 return new ReflectionValue(result);
-            }, 2, 6);
```

Comment:
This represents a substantial change to `Distance()` function that is exposed to users.
- The old version of the function allowed users to calculate distance between any two systems by recording either the names of those two systems or the coordinates of those two systems.
- This version reduces that functionality and may break existing implementations which pass two system names into the function. The new version no longer accepts two system names. Instead, it accepts just one system name and calculates the distance between the named system and the current system.

Rather than removing the two system name option entirely, I think that we should preserve it and add a single system name option for calculating distances between a named system and the current system.

Reply:
You need to look again...

Apology accepted. ;-)

Reply:
Ahh, I see what you did better now. I'll blame that one on drowsiness. ;-) 👍